File: GXpert/GXpert.Web/Modules/Settings/Storage/StorageColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Settings.Columns;

[ColumnsScript("Settings.Storage")]
[BasedOnRow(typeof(StorageRow), CheckNames = true)]
public class StorageColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string AccessKeyId { get; set; }
    public string SecretAccessKey { get; set; }
    public string Endpoint { get; set; }
    public string Region { get; set; }
    public string BucketName { get; set; }
    public short S3Types { get; set; }
    public bool IsDefault { get; set; }
    public string UrlPreview { get; set; }
    public string UploadPath { get; set; }
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Settings/StorageColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { StorageRow } from "./StorageRow";

export interface StorageColumns {
    Id: Column<StorageRow>;
    AccessKeyId: Column<StorageRow>;
    SecretAccessKey: Column<StorageRow>;
    Endpoint: Column<StorageRow>;
    Region: Column<StorageRow>;
    BucketName: Column<StorageRow>;
    S3Types: Column<StorageRow>;
    IsDefault: Column<StorageRow>;
    UrlPreview: Column<StorageRow>;
    UploadPath: Column<StorageRow>;
}

export class StorageColumns extends ColumnsBase<StorageRow> {
    static readonly columnsKey = 'Settings.Storage';
    static readonly Fields = fieldsProxy<StorageColumns>();
}